dbms/src/Server/Server.cpp

```diff
@@ -1267,6 +1268,28 @@ int Server::main(const std::vector<std::string> & /*args*/)
             settings.elastic_threadpool_init_cap,
             std::chrono::milliseconds(settings.elastic_threadpool_shrink_period_ms));
 
+    // For test mode, TaskScheduler is controlled by test case.
+    bool enable_pipeline = settings.enable_pipeline && !global_context->isTest();
+    if (enable_pipeline)
+    {
+        size_t task_executor_thread_num = settings.pipeline_task_executor_threads == 0
+            ? std::thread::hardware_concurrency()
```

Comment:
maybe `getNumberOfLogicalCPUCores()` is better?

Reply:
ok, updated.